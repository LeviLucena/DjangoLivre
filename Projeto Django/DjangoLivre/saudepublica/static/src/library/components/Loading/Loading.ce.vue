<template>
  <div
    v-if="progress"
    ref="root"
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    :progress="progress"
    class="br-loading"
    :data-progress="percent"
  >
    <div class="br-loading-mask full">
      <div class="br-loading-fill"></div>
    </div>
    <div class="br-loading-mask">
      <div class="br-loading-fill"></div>
    </div>
  </div>

  <div
    v-else
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    :style="{ display: 'block', textAlign: 'center' }"
  >
    <div class="loading" :class="{ medium: medium }"></div>
    <span class="rotulo">{{ label }}</span>
  </div>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"

const Loading = {
  name: "brLoading",
  mixins: [TooltipMixin],
  props: {
    /**
     * **[OPCIONAL]** Elemento visual que representa dinamicamente a progressão percentual de 0% a 100%.
     */
    progress: {
      default: false,
      type: Boolean,
    },
    /**
     * **[OPCIONAL]** Elemento visual que representa dinamicamente a progressão percentual de 0% a 100%.
     */
    percent: {
      default: 100,
      type: Number,
      validator: (value) => value >= 0 && value <= 100,
    },
    /**
     * **[OPCIONAL]** Disponível apenas no Loading Indeterminado. Restrito para uso dentro de outros componentes, onde o espaço é menor ou em situações em que se quer dar pouco destaque ao processamento. Também pode ser utilizado com ou sem o rótulo.
     */
    medium: {
      default: false,
      type: Boolean,
    },
    /**
     * **[OPCIONAL]** Elemento visual usado apenas no Loading Indeterminado. Recomenda-se o uso em situações em que a probabilidade de demora do processamento do sistema seja maior.
     */
    label: {
      type: String,
      default: "",
    },
  },
}

export default Loading
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/loading/_loading";

.loading.medium {
  margin-bottom: 16px;
}
</style>
