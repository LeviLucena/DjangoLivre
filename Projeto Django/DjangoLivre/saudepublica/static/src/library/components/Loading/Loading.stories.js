import { kebabiseArgs } from '../../util/Utils.js'
import BrLoading from './Loading.ce.vue'

export default {
  title: 'Componentes/Loading',
  component: BrLoading,
  argTypes: {
    percent: {
      control: {
        type: 'range',
        min: 0,
        max: 100,
        step: 1,
      },
    },
    progress: {
      defaultValue: false,
    },
    medium: {
      defaultValue: false,
    },
    label: {
      defaultValue: '',
    },
    tooltipText: {
      control: 'text',
    },
    tooltipPlace: {
      options: ['top', 'right', 'bottom', 'left'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipType: {
      options: ['info', 'success', 'warning', 'error'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipTimer: {
      control: 'number',
      if: { arg: 'tooltipText', truthy: true },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-loading v-bind="args"></br-loading>`,
})

export const Progress = Template.bind({})
Progress.args = {
  progress: true,
  percent: 10,
}

export const Indeterminate = Template.bind({})
Indeterminate.args = {
  label: 'Carregando...',
}

export const IndeterminateMedium = Template.bind({})
IndeterminateMedium.args = {
  label: 'Carregando...',
  medium: true,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  label: 'Carregando...',
  tooltipText: 'Lorem ipsum dolor sit amet.',
}
