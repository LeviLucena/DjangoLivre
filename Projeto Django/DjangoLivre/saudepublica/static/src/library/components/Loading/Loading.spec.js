import { shallowMount } from '@vue/test-utils'
import Loading from './Loading.ce.vue'

describe('br-loading', () => {
  test('Testa se o componente com a barra de progresso esta renderizando.', () => {
    const wrapper = shallowMount(Loading, {
      propsData: {
        progress: true,
        percent: 80,
      },
    })
    expect(wrapper.find('.br-loading-mask').exists()).toBe(true)
    expect(wrapper.attributes('data-progress')).toBe('80')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa se o componente esta renderizando o loading circular do tipo medium', () => {
    const wrapper = shallowMount(Loading, {
      propsData: { medium: true },
    })
    expect(wrapper.find('.loading').exists()).toBe(true)
    expect(wrapper.find('.medium').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa se o componente esta renderizando o loading circular do tipo pequeno', () => {
    const wrapper = shallowMount(Loading)
    expect(wrapper.find('.loading').exists()).toBe(true)
    expect(wrapper.find('.medium').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa se o componente esta renderizando o label do loading circular', () => {
    const wrapper = shallowMount(Loading, {
      propsData: { rotulo: 'Carregando ...' },
    })
    expect(wrapper.find('.loading').exists()).toBe(true)
    expect(wrapper.find('.rotulo').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Testa se a propriedade de porcentagem foi informada corretamente', () => {
    const wrapper = shallowMount(Loading, {
      propsData: { medium: true, percent: 100 },
    })

    expect(wrapper.find('.loading').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()

    // Extract the validator
    const validator = wrapper.vm.$options.props.percent.validator

    // invoke the validator
    expect(validator(wrapper.vm.$props.percent)).toBe(true)
  })
})
