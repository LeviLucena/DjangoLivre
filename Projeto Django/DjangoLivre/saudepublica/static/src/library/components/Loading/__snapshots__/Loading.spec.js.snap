// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`br-loading Testa se a propriedade de porcentagem foi informada corretamente 1`] = `
<div
  style="display: block; text-align: center;"
>
  <div
    class="loading medium"
  />
  <span
    class="rotulo"
  />
</div>
`;

exports[`br-loading Testa se o componente com a barra de progresso esta renderizando. 1`] = `
<div
  class="br-loading"
  data-progress="80"
  progress="true"
>
  <div
    class="br-loading-mask full"
  >
    <div
      class="br-loading-fill"
    />
  </div>
  <div
    class="br-loading-mask"
  >
    <div
      class="br-loading-fill"
    />
  </div>
</div>
`;

exports[`br-loading Testa se o componente esta renderizando o label do loading circular 1`] = `
<div
  rotulo="Carregando ..."
  style="display: block; text-align: center;"
>
  <div
    class="loading"
  />
  <span
    class="rotulo"
  />
</div>
`;

exports[`br-loading Testa se o componente esta renderizando o loading circular do tipo medium 1`] = `
<div
  style="display: block; text-align: center;"
>
  <div
    class="loading medium"
  />
  <span
    class="rotulo"
  />
</div>
`;

exports[`br-loading Testa se o componente esta renderizando o loading circular do tipo pequeno 1`] = `
<div
  style="display: block; text-align: center;"
>
  <div
    class="loading"
  />
  <span
    class="rotulo"
  />
</div>
`;
